### PR TITLE
feat(op-e2e): enable tx forwarding in sentry nodes

### PR DESCRIPTION
**Pull request by hamdiallam**
By default, transactions can be forwarded as `RollupSequencerHTTP` is setup for the sentry nodes in op-e2e. If desired, this can be disabled through the system config `DisableTxForwarder`

### CODE REVIEW

**Comment by hamdiallam** [comment]
* **#11671** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#11857** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11857?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hamdiallam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: op-e2e/setup.go [hunk 1]
```diff
@@ -624,12 +628,20 @@ func (cfg SystemConfig) Start(t *testing.T, _opts ...SystemConfigOption) (*Syste
...
+	// Ordered such that the Sequencer is initialized first. Setup this way so that
+	// the `RollupSequencerHTTP` GethOption can be supplied to any sentry nodes.
+	l2Nodes := []string{RoleSeq, RoleVerif}
+	for _, name := range l2Nodes {
 		var ethClient services.EthInstance
 		if cfg.ExternalL2Shim == "" {
+			if name != RoleSeq && !cfg.DisableTxForwarder {
+				cfg.GethOptions[name] = append(cfg.GethOptions[name], func(ethCfg *ethconfig.Config, nodeCfg *node.Config) error {
+					ethCfg.RollupSequencerHTTP = sys.EthInstances[RoleSeq].UserRPC().RPC()
+					cfg.Loggers[name].Warn("URL OPTION SET", "url", ethCfg.RollupSequencerHTTP)
```
Comment (tynes) [inline op-e2e/setup.go]: The `if` statement is hard to parse due to the amount of boolean logic in it. If its not the sequencer and its not disable tx forwarder, this is saying if its a verifier and forwarding is enabled. Just typing this out to double check. I also think including the role name in the logline is nice

File: op-e2e/setup.go [hunk 2]
```diff
@@ -638,18 +650,25 @@ func (cfg SystemConfig) Start(t *testing.T, _opts ...SystemConfigOption) (*Syste
...
 			}
+
 			ethClient = l2Geth
 		} else {
 			if len(cfg.GethOptions[name]) > 0 {
 				t.Skip("External L2 nodes do not support configuration through GethOptions")
 			}
+
+			if name != RoleSeq && !cfg.DisableTxForwarder {
+				cfg.Loggers[name].Warn("External L2 nodes do not support `RollupSequencerHTTP` configuration. No tx forwarding support.")
```
Comment (tynes) [inline op-e2e/setup.go]: This could `panic` to prevent misconfiguration to ensure that people don't think they are doing something that they cannot, not sure if that would break many tests

File: op-e2e/setup.go (hunk 1 above)
Comment (hamdiallam) [reply op-e2e/setup.go]: yea that's right. Also the log line was included by accident when debugging why this code change wasn't working. I'm going to remove it as I dont think it's particularly useful

File: op-e2e/setup.go (hunk 2 above)
Comment (hamdiallam) [reply op-e2e/setup.go]: Since `DisableTxForwarder` is `true` by default, this would panic by default as well hence the warning instead of erroring out.

It felt more appropriate to enable tx forwarding by default instead of explicitly enabling it. do you think otherwise?

File: op-e2e/setup.go [hunk 3]
```diff
@@ -624,12 +628,19 @@ func (cfg SystemConfig) Start(t *testing.T, _opts ...SystemConfigOption) (*Syste
...
 	}
 
-	for name := range cfg.Nodes {
-		if name == RoleL1 {
-			return nil, fmt.Errorf("node name %s is reserved for L1 node", RoleL1)
-		}
+	// Ordered such that the Sequencer is initialized first. Setup this way so that
+	// the `RollupSequencerHTTP` GethOption can be supplied to any sentry nodes.
+	l2Nodes := []string{RoleSeq, RoleVerif}
+	for _, name := range l2Nodes {
```
Comment (protolambda) [inline op-e2e/setup.go]: If someone added a node to the `cfg.Nodes` then that nodes will not be started anymore, seems like a bug.
Maybe you can set `l2Nodes := maps.Keys(cfg.Nodes)`, and then sort the `l2Nodes`, with a sort-func that puts the `RoleSeq` first?

**Review by protolambda — COMMENTED** [review]
Looks good, one concern about the nodes config iteration though

File: op-e2e/setup.go (hunk 3 above)
Comment (hamdiallam) [reply op-e2e/setup.go]: yea good call. Let me update

**Review by tynes — APPROVED** [review]
